File: app/src/main/java/com/yknx4/notificationtracker/network/endpoints/EchoService.java
package com.yknx4.notificationtracker.network.endpoints;

import com.yknx4.notificationtracker.API;
import com.yknx4.notificationtracker.Constants;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

/**
 * Created by yknx4 on 7/14/16.
 */

public interface EchoService {
    @GET(API.ECHO_ENDPOINT)
    Call<String> index();

    @POST(API.ECHO_ENDPOINT)
    Call<String> create(@Body String body);

}

File: app/src/main/java/com/yknx4/notificationtracker/network/endpoints/EchoService.kt
package com.yknx4.notificationtracker.network.endpoints

import com.google.gson.JsonElement
import com.google.gson.JsonObject
import com.yknx4.notificationtracker.API
import com.yknx4.notificationtracker.Constants

import retrofit2.Call
import retrofit2.http.*

/**
 * Created by yknx4 on 7/14/16.
 */

interface EchoService {
    @GET(API.ECHO_ENDPOINT)
    fun index(): Call<String>

    @Headers("Content-type: application/json")
    @POST(API.ECHO_ENDPOINT)
    fun create(@Body body: JsonElement): Call<JsonElement>

}
